### PR TITLE
404 page redirects fitting vanity paths to the long-form url

### PR DESCRIPTION
**Pull request by DriftingSands**
404 page now checks if the current path could be a vanity path for an existing path that was not re-routed

Test URLs:
- Before: https://main--helix-sportsmagazine--headwirecom.hlx.page/story/phil-mickelson-tweet-pga-tour-liv-merger
- After: https://404-vanity-redirect--helix-sportsmagazine--headwirecom.hlx.page/story/phil-mickelson-tweet-pga-tour-liv-merger

### CODE REVIEW

File: 404.html [hunk 1]
```diff
@@ -179,12 +179,22 @@ <h1>TAKE A MULLIGAN</h1>
...
-      const pictureElement = createOptimizedPicture(imageUrl, imageAlt);
-      const imageContainer = document.querySelector(".error-image");
-      imageContainer.appendChild(pictureElement);
+      import { createOptimizedPicture } from "/scripts/lib-franklin.js";
+      import sitemap from "/import/tools/importer/data/sitemap.json" assert {type: 'json'}
+
+      const pathLocation = sitemap.find((obj) => Object.hasOwn(obj, window.location.pathname))
+      if (pathLocation) {
+        const drivePath = '/content-v2/' + pathLocation[window.location.pathname].replace('/content/golfdigest-com/en/', '').replace('.html', '')
+        window.location.assign(drivePath)
```
Comment (icaraps) [inline 404.html:188]: If the user clicks on the back button, does he end up in an infinite loop in this case ?

File: 404.html [hunk 2]
```diff
@@ -179,12 +179,22 @@ <h1>TAKE A MULLIGAN</h1>
...
     <footer></footer>
     <script type="module">
-      import { createOptimizedPicture } from "./scripts/lib-franklin.js";
-      const imageUrl = "/media_10b8f0c21a329ee5ab764abf34d64a94de757ea2a.png";
-      const imageAlt = "Tiger Woods regretting a bad shot";
-      const pictureElement = createOptimizedPicture(imageUrl, imageAlt);
-      const imageContainer = document.querySelector(".error-image");
-      imageContainer.appendChild(pictureElement);
+      import { createOptimizedPicture } from "/scripts/lib-franklin.js";
+      import sitemap from "/import/tools/importer/data/sitemap.json" assert {type: 'json'}
```
Comment (icaraps) [inline 404.html:183]: That doesn't work on firefox and other browsers see https://caniuse.com/mdn-javascript_statements_import_import_assertions_type_json

**Review by icaraps — CHANGES_REQUESTED** [review]

File: 404.html (hunk 1 above)
Comment (DriftingSands) [reply 404.html:188]: no, I have not found that to be the case

File: 404.html (hunk 2 above)
Comment (DriftingSands) [reply 404.html]: firefox should now be supported

File: scripts/scripts.js [hunk 3]
```diff
@@ -30,6 +30,17 @@ const LCP_BLOCKS = [...Object.values(ARTICLE_TEMPLATES), 'hero']; // add your LC
 
 const range = document.createRange();
 
+// getting real path, and adjusting canonical link to use the vanity path
+const canonicalLinkTag = document.head.querySelector('link[rel="canonical"]');
+if (canonicalLinkTag) {
+  const longPathMetadata = document.createElement('meta');
+  longPathMetadata.setAttribute('property', 'hlx:long-form-path');
```
Comment (icaraps) [inline scripts/scripts.js:37]: @DriftingSands  Same comment as https://github.com/headwirecom/helix-sportsmagazine/pull/269/files#r1339544494

**Pull request merged**